lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1367,6 +1367,16 @@ LLVMType LLVMType::getFloatTy(LLVMDialect *dialect) {
 LLVMType LLVMType::getHalfTy(LLVMDialect *dialect) {
   return dialect->impl->halfTy;
 }
+LLVMType LLVMType::getQuadTy(LLVMDialect *dialect) {
+  return getLocked(dialect, [=] {
```

Comment:
These types are thread-safe to query from the LLVMContext, so we can cache them as we do the others.

Reply:
Let's also use the same method name as the LLVM::get methods if possible.

Reply:
OK, I wasn't sure if it was better to cache them or not.

Reply:
I added caching for most of the ones that makes sense(mostly mirroring what happens within LLVMContextImpl internally). We mainly try to avoid calling getLocked here if we can given that it will lock two mutexs: 1 for the LLVM module, and 1 internally in the MLIR context.

Reply:
Thank you for the explanation.